packages/flutter/lib/src/widgets/form.dart

```diff
@@ -369,7 +370,7 @@ class FormState extends State<Form> {
       }
     }
 
-    if (errorMessage.isNotEmpty) {
+    if (errorMessage.isNotEmpty && MediaQuery.announceOf(context)) {
```

Comment:
I see @Piinks already approved this but given this was reverted once I might suggest breaking this pr into announce changes and "widgets that use announce changes". 

Not required but something to consider.